lib/storage/metadata/mongoclient/MongoClientInterface.ts

```diff
@@ -859,45 +871,71 @@ class MongoClientInterface {
             ordered: true,
         })
             .then(() => cb(null, `{"versionId": "${versionId}"}`))
-            .catch((err) => {
+            .catch((err: MongoBulkWriteError) => {
                 /*
                  * Related to https://jira.mongodb.org/browse/SERVER-14322
                  * It happens when we are pushing two versions "at the same time"
                  * and the master one does not exist. In MongoDB, two threads are
                  * trying to create the same key, the master version, and one of
                  * them, the one with the highest versionID (less recent one),
                  * fails.
-                 * We check here than than the MongoDB error is related to the
-                 * second operation, the master version update and than the error
-                 * code is the one related to mentionned issue.
                  */
                 if (err.code === 11000) {
                     log.debug('putObjectVerCase1: error putting object version', {
                         code: err.code,
                         error: err.errmsg,
-                        isRetry: isRetry ? true : false, // eslint-disable-line no-unneeded-ternary
+                        isRetry: isRetry ? true : false,
```

Comment:
nit: the ternary is only needed (is it really?) for the case where isRetry is undefined [based on the type], so we may do this instead
```suggestion
                        isRetry: isRetry || false,
```
or if we want to be even more resilient:
```suggestion
                        isRetry: !!isRetry,
```